### ISSUE TITLE
CNNEncoder2 好象不能使用三维的图片，只能改用CNNEncoder3

### ISSUE THREAD

**Issue by gg22mm** [xianlin7/ConvFormer#5]
CNNEncoder2 好象不能使用三维的图片，只能改用CNNEncoder3，  用的是 分割Patch的方式做的吧？
三维的图像不行吗？
![Q}Z6R}`YW$XM OV}JC2S%KD](https://github.com/xianlin7/ConvFormer/assets/6648329/0a0940d2-ba19-4f18-93b6-8518e2adbe36)

![X~4N8K~W2U9EVHIHR% B$9V](https://github.com/xianlin7/ConvFormer/assets/6648329/00648dde-0519-46eb-b4b0-c8994b049a89)
  这个代码是直接构建代码吗？ 感觉不象呀，看到了 Rearrange 分割图像

**Comment by gg22mm** [comment]
我换了下面的方式，虽然没有使用Rearrange 分割图像，但是效果比较差，图片尺寸：256*256 可以用下面卷积： 
这种方式还不如上同用的 Rearrange 分割图像 。



    class CNNEncoder2(nn.Module):
    def __init__(self, n_channels, out_channels, patch_height, patch_width):
        super().__init__()

        # SingleConv         
        self.inc = nn.Sequential(
            nn.Conv2d(n_channels, 64, kernel_size=3, padding=1, bias=False),
            nn.BatchNorm2d(64),
            nn.ReLU(inplace=True)
        )

        # DownSingleConv
        self.down = nn.Sequential(
            # 1
            nn.MaxPool2d(2),
            nn.Conv2d(64, 128, kernel_size=3, padding=1, bias=False),
            nn.BatchNorm2d(128),
            nn.ReLU(inplace=True),

            # 2
            nn.MaxPool2d(2),
            nn.Conv2d(128, 256, kernel_size=3, padding=1, bias=False),
            nn.BatchNorm2d(256),
            nn.ReLU(inplace=True),           

            # 3
            nn.MaxPool2d(2),
            nn.Conv2d(256, 512, kernel_size=3, padding=1, bias=False),
            nn.BatchNorm2d(512),
            nn.ReLU(inplace=True),

            # 4
            nn.MaxPool2d(2),
            nn.Conv2d(512, out_channels, kernel_size=3, padding=1, bias=False),
            nn.BatchNorm2d(out_channels),
            nn.ReLU(inplace=True),
        )        

    def forward(self, x):
        x = self.inc(x)
        x = self.down(x)       
        return x

**Comment by LambdaLi** [comment]
请问3D可行吗？

**Comment by gg22mm** [comment]
我没试，应该可以，主打就是3D ~~
